### DISCUSSION TITLE
Is it possible to use `bevy_reflect` with (static) string references as struct fields?

### DISCUSSION THREAD

**Discussion by mainrs** [bevyengine/bevy#14184] [Q&A]
For example

```rust
#[derive(Reflect)]
pub struct MyStruct<'a> {
    pub id: &'a str, // 'a this static in my case. Just the definition is using a variable lifetime.
}
```

**Comment by MrGVSV** [comment]
Yes. `bevy_reflect` is implemented for `&'static str`. And the derive macro will only generate an impl to cover cases where `'a: 'static`. So it should work.

**Edit:** And to clarify I mean the derive macro will generate something like this:

```rust
impl<'a> Reflect for MyStruct<'a>
where
  Self: 'static + Send + Sync,
  // ...
{
  // ...
}
```

That `Self: 'static` clause means that `MyStruct<'a>` will be `Reflect`, but only when it is `'static` (i.e. `MyStruct<'static>`).
